packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts

```diff
@@ -0,0 +1,143 @@
+import BigNumber from 'bignumber.js'
+
+export function getExponentValue(decimals = 0): BigNumber {
+    return new BigNumber(10).pow(decimals)
+}
+
+export function getHumanValue(value?: BigNumber, decimals = 0): BigNumber | undefined {
+    return value?.div(getExponentValue(decimals))
+}
+
+export function getNonHumanValue(value: BigNumber | number, decimals = 0): BigNumber {
+    return new BigNumber(value).multipliedBy(getExponentValue(decimals))
+}
+
+export function getGasValue(price: number): number {
+    return getNonHumanValue(price, 9).toNumber()
+}
+
+export function formatBigValue(
+    value?: BigNumber | number,
+    decimals = 4,
+    defaultValue = '-',
+    minDecimals: number | undefined = undefined,
+): string {
+    if (value === undefined) {
+        return defaultValue
+    }
+
+    const bnValue = new BigNumber(value)
+
+    if (bnValue.isNaN()) {
+        return defaultValue
+    }
+
+    return new BigNumber(bnValue.toFixed(decimals)).toFormat(minDecimals)
+}
+
+type FormatNumberOptions = {
+    decimals?: number
+}
+
+export function formatNumber(value: number | BigNumber | undefined, options?: FormatNumberOptions): string | undefined {
+    if (value === undefined || Number.isNaN(value)) {
+        return undefined
+    }
+
+    const { decimals } = options ?? {}
+
+    const val = BigNumber.isBigNumber(value) ? value.toNumber() : value
```

Comment:
if this a big number over 53 bits?

see https://tc39.es/ecma262/#sec-ecmascript-language-types-number-type

Reply:
I don't believe it is so we should be OK